org.eclipse.m2e.jdt/src/org/eclipse/m2e/jdt/internal/ModuleSupport.java

```diff
@@ -682,14 +698,27 @@ public List<JpmsArgValue> getFromArgs(JpmsArgs args) {
      * @return
      */
     static JpmsArgType valueFromArgumentName(String argumentName) {
+      if(argumentName == null) {
+        return null;
+      }
+
       Optional<JpmsArgType> result = Arrays.stream(JpmsArgType.values())
           .filter(at -> at.getArgumentName().equalsIgnoreCase(argumentName)).findFirst();
 
-      if(result.isEmpty()) {
-        return null;
+      if(!result.isEmpty()) {
+        return result.get();
       }
 
-      return result.get();
+      // try to match argument with delimiter
+      result = Arrays.stream(JpmsArgType.values())
+          .filter(at -> argumentName.toLowerCase().startsWith(at.getArgumentName().toLowerCase() + argumentDelimiter))
+          .findFirst();
+
+      if(!result.isEmpty()) {
+        return result.get();
+      }
```

Comment:
```suggestion
      return result.or(() -> Arrays.stream(JpmsArgType.values())
          .filter(at -> argumentName.toLowerCase().startsWith(at.getArgumentName().toLowerCase() + argumentDelimiter))
          .findFirst()).orElse(null);
```

Reply:
let me create a new PR to refactor it.

Reply:
No need to if you're busy with other things; it's more food for thoughts than an actual requirement or request.